tutorials/basic-topics/linear-search.md

```diff
@@ -0,0 +1,174 @@
+---
+title: 'Linear search'
+description: 'Linear search is a searching technique that works in a linear/sequential fashion. It traverses whole of the data structure and returns the position(s) where the value is found. If not found then it simple returns nothing.'
+hide_table_of_contents: true
+---
+
+<TutorialAuthors names="@siddoinghisjob"/>
+
+## Overview
+
+Lets say we have a linear data structure - array, linked list - and we need to search for a certain element. To search there are mainly two searching techniques, one being linear search. In linear search we traverse the whole array and then while traversing we check for the particular item. If there's a match then we return that position( we can return multiple positions too).
+
+Example:
+``` 
+  Lets have an array [1,2,6,5,6] and we need to find 6. 
+  Then we can traverse the array from index 0 to index n-1 i.e., 4 here and print the positions of 6. 
+      Index - Element - Output
+        0        1           
+        1        2
+        2        6        2
+        3        5          
+        4        6        4
+        
+   Hence the final output will be - 2 4
+```
+To achive this, we need to employ a for loop and traverse over the entire array/linked list. And check each element of the data structure for the desired value. If found then we will output that value. This will take time complexity of **O(n)** i.e., it will take `linear` time complexity as we have run the for loop once to traverse all elements.
+
+Code in c++ :-
+
+<Tabs>
+
+<TabItem value="cpp" label="C++">
+<SolutionAuthor name="@siddoinghisjob"/>
+
+```cpp
+#include<iostream>
+using namespace std;
+
+int main(){
+  // Array to keep the values
+  int arr[5] = {1,2,3,4,5};
+  // Item to be searched 
+  int item = 3; 
+  for(int i = 0; i < 5; i++){
+    // checking if item is arr[i]
+    if(arr[i] == item){  
+      cout << "Found at " << i << endl;
+    }
+  }
+} 
```

Comment:
I've already mentioned that you can only use LC problem to walk through. Random example is not accepted.